src/Geom_Poly.f90

```diff
@@ -361,20 +373,141 @@ ELEMENTAL FUNCTION isCircle_PolygonType(thisPoly) RESULT(bool)
     ENDFUNCTION isCircle_PolygonType
 !
 !-------------------------------------------------------------------------------
+!> @brief Determines whether a polygon meets the criteria for being an arc
+!>        sector. In short, the polygon must have exactly two quadratic edges
+!>        and exactly two non-quadratic edges. A circle is not an arc sector.
+!> @param thisPoly The polygon type to query
+!> @param bool The logical result of the operation
+!>
+    ELEMENTAL FUNCTION isSector_PolygonType(thisPoly) RESULT(bool)
+      CLASS(PolygonType),INTENT(IN) :: thisPoly
+      LOGICAL(SBK) :: bool
+      INTEGER(SIK) :: nNonQuadEdges!,i
+      ! REAL(SRK) :: dist,dist_old
+      bool=.FALSE.
+      IF(thisPoly%isinit) THEN
+        !nVert should be equal to the total number of edges
+        nNonQuadEdges=thisPoly%nVert-thisPoly%nQuadEdge
+        bool=(thisPoly%nQuadEdge == 2) .AND. (nNonQuadEdges == 2)
```

Comment:
If above is DBC will this execute w/o segfault if the object is not initialized

Reply:
Yes, it should execute but I will try adding DBC because it should be a requirement that it is initialized.